src/shared/elf-util.c

```diff
@@ -532,14 +532,21 @@ static int module_callback(Dwfl_Module *mod, void **userdata, const char *name,
         for (size_t i = 0; i < n_program_headers; ++i) {
                 GElf_Phdr mem, *program_header;
                 Elf_Data *data;
+                GElf_Addr end_of_segment;
 
                 /* The core file stores the ELF files in the PT_LOAD segment. */
                 program_header = sym_gelf_getphdr(elf, i, &mem);
                 if (!program_header || program_header->p_type != PT_LOAD)
                         continue;
 
+                /* Check that the end of segment is a valid address. */
+                if (__builtin_add_overflow(program_header->p_vaddr, program_header->p_memsz, &end_of_segment)) {
```

Comment:
iirc C semantics mandate that having a pointer to invalid memory is undefined behaviour, no matter what.

Thus, please don't print the address via %p here. Just output it as hex integer...

Reply:
Ok, so now both format string use `%#zx` with an explicit cast to `size_t` when passing the arguments.